### ISSUE TITLE
nconf.env() loads everything with the default separator ":" even when logicalSeparator is specified

### ISSUE THREAD

**Issue by tleunen** [indexzero/nconf#301]
As the title says, `nconf.env()` doesn't care about `logicalSeparator` because it uses `common.key` which is hardcoded to use `:`. As seen here: https://github.com/indexzero/nconf/blob/master/lib/nconf/common.js#L31-L33
```js
//
// ### function key (arguments)
// Returns a `:` joined string from the `arguments`.
//
common.key = function () {
  return Array.prototype.slice.call(arguments).join(':');
};
```

**Comment by mhamann** [comment]
@tleunen can you provide a reproduction of this issue, please? I use this library in a variety of projects, and can attest that `nconf.env({ separator: '__' })` works perfectly.

**Comment by tleunen** [comment]
Quickly, doing this doesn't work to go from an env var `XX__YY` to `XX.YY`. It will generate `XX:YY` without using the logical separator specified.

```js
nconf.env({
  separator: '__',
  logicalSeparator: '.'
});
```

**Comment by thewizarodofoz** [comment]
@mhamann this is quite important for me, since I am working with Kubernetes, and config keys can only use `-` `_` `,` but not `:`.

**Comment by mhamann** [comment]
@thewizarodofoz in k8s, you'd set an env var with something like this: `my__env__var`,which could then be accessed like `my:env:var`. That works, but I think the issue you're seeing is that you want to access it via a different separator, e.g. `my.env.var`?

**Comment by thewizarodofoz** [comment]
@mhamann sorry I was too vauge.
I am injecting env into pods in k8s using [configmap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/).
So let's say I want to pass this env to nconf:
```bash
namespaced:a=a
namespaced:b=b
```
and then get `namespaced` as an object, I would do:
```js
nconf.env()
nconf.get('namespaced') // {a: 'a', b: 'b'}
```
Problem is, [valid env keys in k8s configmaps](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#configmap-v1-core) must consist of alphanumeric characters, `-`, `_` or `.`. So `namespaced:a` is not a valid key for k8s configmap. I would like to be able to tell nconf to use `.` as a logical separator like this:
```js
nconf.env({logicalSeparator: '.'})
nconf.get('namespaced') // {a: 'a', b: 'b'}
```

Does this make sense?

**Comment by thewizarodofoz** [comment]
@mhamann Sorry I misunderstood this issue. Please ignore my comments.

**Comment by mhamann** [comment]
@tleunen here's an alternative solution. Feel free to share your thoughts: https://github.com/indexzero/nconf/pull/337

**Comment by mhamann** [comment]
#337 will be released as part of v1.0.0, so closing this one as resolved. Let me know if you feel differently.
